src/Core/Domain/Category/ValueObject/CategoryId.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Category\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryException;
+
+/**
+ * Class CategoryId.
+ */
+class CategoryId
+{
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @param int $id
+     *
+     * @throws CategoryException
+     */
+    public function __construct($id)
+    {
+        $this->setId($id);
+    }
+
+    /**
+     * @return int
+     */
+    public function getValue()
```

Comment:
You're in an id context, you can have a simple get and set method. Otherwise you have a setId, so you must be consistent and have a getId method. Or setValue and getValue.

Reply:
as i understand it, its nothing wrong with it, because:
1. `set` method would be incorect, because if you need to change value of `CategoryId`, you better create new instance of it.
2. regarding `getId`, should it return `CategoryId` or it's value? because we already have an id, all we want to do is get it's value.

i may be wrong, but i think @eternoendless can give more insights on this. :+1:

Reply:
In fact creating a new instance will increase memory usage for nothing :/
thinking about a simple property style with get / set methods.

Reply:
yup, but value objects should as well be immutable. :/